tests/SprykerSdkTest/AsyncApi/Console/SchemaMessageAddConsoleTest.php

```diff
@@ -100,4 +101,67 @@ public function testAddMessageReturnsErrorCodeAndPrintsErrorMessagesWhenMessageT
             ['verbosity' => OutputInterface::VERBOSITY_VERBOSE],
         );
     }
+
+    /**
+     * This test ensures that a message can only be added once per channel.
+     * The given API file already has the message that should be added and thus adding it again will be skipped.
+     *
+     * @return void
+     */
+    public function testAddMessageOnlyWhenMessageNameDoesNotExistInChannel(): void
+    {
+        // Arrange
+        $commandTester = $this->tester->getConsoleTester(SchemaMessageAddConsole::class, false);
+
+        // Act
+        // The message 'OutgoingMessage' is added for the second time in the channel 'channelNameA'
+        $commandTester->execute(
+            [
+                '--' . SchemaMessageAddConsole::OPTION_MESSAGE_TYPE => 'subscribe',
+                '--' . SchemaMessageAddConsole::OPTION_PROPERTY => ['property:string'],
+                '--' . SchemaMessageAddConsole::OPTION_ASYNC_API_FILE => codecept_data_dir('api/asyncapi/asyncapi.yml'),
+                SchemaMessageAddConsole::ARGUMENT_CHANNEL_NAME => 'channelNameA',
+                SchemaMessageAddConsole::ARGUMENT_OPERATION_ID => 'operationId',
+                SchemaMessageAddConsole::ARGUMENT_MESSAGE_NAME => 'OutgoingMessage',
+            ],
+        );
+
+        // The YAML file is parsed after the message was added
+        $asyncApi = Yaml::parseFile(codecept_data_dir('api/asyncapi/asyncapi.yml'));
+
+        // Assert
+        // The message exists only once in the channel
+        $this->tester->assertMessageExistsOnlyOnceInChannel($asyncApi, 'OutgoingMessage', 'channelNameA', 'subscribe');
+    }
+
+    /**
+     * This test ensures that the array 'oneOf' is not created when trying to add an existing message to a channel that has only one message.
+     *
+     * @return void
+     */
+    public function testAddExistingMessageToChannelWithOneMessageDoesNotCreateOneOfArray(): void
```

Comment:
Isn't this exactly the same test as the one before? Both test that a message can only be added once. That we create `oneOf` when we have more than one message is not of interest for this bug.

Or am I getting it wrongly?

Reply:
I had to add this test to cover this part of `AsyncApiBuilder`:
```
if ($this->messageNameExists($messageName, $messages) === true) {
    return $asyncApi;
}
```
This show that if the array `oneOf` still doesn't exist and the message is not added, the array is not created unnecessarily.

Reply:
Then it should stay when its for coverage reasons, the diff between both tests is not obvious. Can you highlight it by a doc block comment maybe?

Reply:
Yes, I know that the difference is fuzzy. Is it ok if I put something like this in the docblock?
```
Coverage:
if ($this->messageNameExists($messageName, $messages) === true) {
    return $asyncApi;
}
```

Reply:
I've changed the description of the method and added the test coverage, I think is much better now.